src/test/scala/edu/ie3/simona/model/participant/HpModelSpec.scala

```diff
@@ -232,55 +232,209 @@ class HpModelSpec
+                Some(hpData.ambientTemperature),
+                Kilowatts(95.0),
+                Kilowatts(80.0),
+                ThermalGridState(
+                  Some(ThermalHouseState(0L, Celsius(22), Kilowatts(80))),
+                  Some(
+                    ThermalStorageState(0L, KilowattHours(500), Kilowatts(0))
+                  ),
+                ),
+                Some(HouseTemperatureUpperBoundaryReached(0L)),
+              ),
+              (0.0, 0.0, 0.0),
+            ),
+
+            // No capacity for flexibility at all when house has been (external) heated up above upperTemperatureBoundary
+            (
+              ThermalGridState(
+                Some(ThermalHouseState(0L, Celsius(25), Kilowatts(0))),
+                Some(ThermalStorageState(0L, KilowattHours(500), Kilowatts(0))),
+              ),
+              HpState(
+                isRunning = false,
+                0,
+                Some(hpData.ambientTemperature),
+                Kilowatts(95.0),
+                Kilowatts(80.0),
+                ThermalGridState(
+                  Some(ThermalHouseState(0L, Celsius(25), Kilowatts(0))),
+                  Some(
+                    ThermalStorageState(0L, KilowattHours(500), Kilowatts(0))
+                  ),
+                ),
+                None,
+              ),
+              (0.0, 0.0, 0.0),
             ),
-          )
-          val lastState = HpState(
-            isRunning = true,
-            0,
-            Some(hpData.ambientTemperature),
-            Kilowatts(95.0),
-            Kilowatts(80.0),
-            thermalState,
-            Some(HouseTemperatureUpperBoundaryReached(7995L)),
           )
 
-          hp.determineFlexOptions(relevantData, lastState) match {
-            case ProvideMinMaxFlexOptions(
-                  modelUuid,
-                  referencePower,
-                  minPower,
-                  maxPower,
-                ) =>
-              modelUuid shouldBe hp.uuid
-              referencePower should approximate(Kilowatts(95.0))
-              minPower should approximate(Kilowatts(0.0))
-              maxPower should approximate(Kilowatts(95.0))
-          }
+        // Run the test cases
+        forAll(testCases) {
+          (
+              thermalState: ThermalGridState,
+              lastState: HpState,
+              expectedValues: (Double, Double, Double),
+          ) =>
+            val (expectedReferencePower, expectedMinPower, expectedMaxPower) =
+              expectedValues
+
+            // Initialize the house and grid models
+            val house =
+              thermalHouse(18, 22).copy(ethLosses = WattsPerKelvin(200))
+            val grid = thermalGrid(house, Some(thermalStorage))
+            val hp = hpModel(grid)
+
+            // Create relevant data for the current test
+            val relevantData = hpData.copy(currentTick =
+              thermalState.houseState.map(_.tick).getOrElse(0L)
+            )
```

Comment:
Last tick is 0 and current tick is 0. This usually would not happen. I saw that you set `activePower` and `qDot` to different values for this test, but these values are ignored anyways, right? Because, to the current state (once calculated), it's not relevant what the last power has been.

So maybe it'd be clearer to use, let's say, 0 as last tick and 900 as current tick, and just set the powers to 0? Looking through the model, on my side there's still some confusion left (relating to our discussion about `calcState`), so maybe we have to get some clarity there beforehand.

Reply:
I will summarise our discussion for future reference: This is correct. As we are only testing the FlexOptions here, we can even use the same tick to keep the test case simple. 
The current state depends on the last power, because that is how we distinguish what the reference behaviour will be in this case. For example: House is between lower and target temperature. If the HP was active in the last state, the house was in the heating phase, so our reference behaviour will still be 'active HP'. If it was not active in the last state, the house is cooling, our reference behaviour should be non-active hp until the house reaches the lower temp boundary.

I will add an explanatory comment.

This might be easier to test / need to be changed once we introduced the transitoryData.